package/run.ps1

```diff
@@ -46,6 +46,40 @@ function Get-StringHash {
     return (Get-FileHash -InputStream $stringAsStream -Algorithm SHA256).Hash.ToLower()
 }
 
+# stops the RKE2 process as well as other processes spawned from rke2
+function Stop-RKE2-Processes () {
+    $rke2ProcessNames = @('rke2', 'kube-proxy', 'kubelet', 'containerd', 'calico-node')
+    foreach ($processName in $rke2ProcessNames) {
+        Write-LogInfo "Checking if $processName process exists"
+        if (Get-Process -Name $processName -ErrorAction SilentlyContinue) {
+            Write-LogInfo "$processName process found, stopping now"
+            Stop-Process -Name $processName
+        }
+    }
+
+    # ensure all processes have exited
+    $waitLimit = 3
+    foreach ($processName in $rke2ProcessNames) {
+        $currentWait = 0
+        while ($waitLimit -ne $currentWait) {
```

Comment:
if we're executing a break on a successful exit and fatal on reaching the limit, maybe we should just make this:

```suggestion
        while ($true) {
```